### DISCUSSION TITLE
Rasterize/Interpolate many PET-POIs in Solweig

### DISCUSSION THREAD

**Discussion by MaryWP** [UMEP-dev/UMEP#381] [Ideas]
Hello,

we had an idea about getting the PET (or UTCI) output of SOLWEIG into a raster format.
We let QGIS produce a set of regular points in a 2 m raster. (However, I will probably use 1m the next time.) Those were the input vector for the POIs in the SOLWEIG calculation.

Then, I wrote a R-Script, which takes out all the PET/UTCI values for a certain time step and puts them into a table (with the point IDs).
After that I get the table into QGIS, join with the regular points (via the IDs) and then interpolate/rasterize (whatever works).

It's an easy workflow, which produced a fast output.

Did somebody try anything similar?
I couldn't find a matching discussion, but would be happy to discuss :)

**Comment by biglimp** [comment]
Hi,
Nice GIS-based approach. We (@j3r3m1 and me) are currently working with the inclusion of a simple wind model to be able to directly put out maps of PET and UTCI. Hopefully we have an experimental version before the summer.

**Reply by elidamd** [reply]
Interesting, I'd like to follow this thread about PET output maps.

**Reply by paula-lelis** [reply]
Hi, I am also interested about this.

**Reply by ElyumOfTheRocks** [reply]
Greetings from Montreal! I'm very interested by that output as well, do you need any testers for your experimental version?

**Comment by paula-lelis** [comment]
Hello, MaryWP, could you please share with me this methodology? I need to create UTCI maps for an entire city and I don't know yet exactly how to proceed using UMEP.

**Reply by MaryWP** [reply]
Hey Paula,
sorry for the late reply.

Before running SOLWEIG your have to create regular points in the extent of your input data. I usually use the extent of my DSM or DEM. QGIS has an own tool for this. The distance between the points should be the resolution of your input data.
Before using the points in SOLWEIG, you have to delete the point close to the edge of the project area (or else SOLWEIG won't be able to run). I usually just use a specific area of interest in the middle of the calculated area.
When running SOLWEIG you just use the regular points as the POIs. It will then save a txt-file for every POI. Now you have to get everything in one csv-File. I use a short code in R. It just takes the data of all the files (of one time step) and puts the PET values and the IDs of the points in one table.
When you are done with this, you get back to QGIS and import the table with IDs and PET values. And then you join the table with the regular points from before (via the point IDs).
After that you just interpolate the data points (i usually use IDW) and you are done.

I hope this helps little!

**Reply by paula-lelis** [reply]
Hello, MaryWP, thank you very much for your reply! I did exactly the same, with a R-script to join the txt files and to filter the relevant information (PET and UTCI values). It took me a while to find out that SOLWEIG do not run if the points are close to the edge...I had the same problem and fixed it. Thank you very much and good luck with your research! Best regards, Paula.

**Reply by elidamd** [reply]
Hi,
I followed your idea, with some customization, and it also worked well.
Because I had hundreds of thousands of POI, I customized the code from solweig to produce the PET values in a single file. Afterward I had to do some manual work separating rows of each hour into different columns. The final part was similar, by joining the PET values in QGIS to generate raster maps.
Thanks a lot for sharing your solution, it was really helpful!
